src/main/java/com/kodilla/ecommercee/controller/UserController.java

```diff
@@ -31,31 +28,22 @@ public UserDto getUser(@RequestParam Long userId) throws UserNotFoundException {
         return userMapper.mapToUserDto(service.getUser(userId).orElseThrow(UserNotFoundException::new));
     }
 
-    @RequestMapping(method = RequestMethod.DELETE, value = "deleteUser")
-    public void deleteUser(@RequestParam Long userId) {
-            service.delete(userId);
-    }
 
-    @RequestMapping(method = RequestMethod.PUT, value = "updateUser")
-    public UserDto updateUser(@RequestBody UserDto userDto) {
+    @RequestMapping(method = RequestMethod.PUT, value = "updateUserDto")
+    public UserDto updateUserDto(@RequestBody UserDto userDto) {
         User user = userMapper.mapToUser(userDto);
         User savedUser = service.saveUser(user);
         return userMapper.mapToUserDto(savedUser);
     }
 
+    @RequestMapping(method = RequestMethod.PUT, value = "updateUser")
+    public User updateUser(@RequestBody User user) {
+        return service.saveUser(user);
+    }
+
     @RequestMapping(method = RequestMethod.PUT, value = "generateUserKey")
-    public UserDto generateUserKey(@RequestParam Long userId) {
-        try {
-            UserDto userToChangeKey = getUser(userId);
-            int key = (int)(Math.random()*20000);
-            userToChangeKey.setUserKey(String.valueOf(key));
-            userToChangeKey.setExpirationTime(LocalDateTime.now().plusHours(1L));
-            User savedUser = service.saveUser(userMapper.mapToUser(userToChangeKey));
-            return userMapper.mapToUserDto(savedUser);
-        } catch (UserNotFoundException e) {
-            System.out.println("User not found");
-            return null;
-        }
+    public void generateUserKey(@RequestParam Long userId) {
```

Comment:
Dobrze by było gdyby ktoś się jeszcze na ten temat wypowiedział ale wydaje mi się, że powinniśmy tutaj uwzględnić jeszcze tego JSONa do logowania, bo w specyfikacji mamy, że endpoint ma wygenerować losowy klucz po podaniu poprawnych danych użytkownika, więc powinniśmy tego użytkownika jakoś zweryfikować. Dlatego wydaje mi się, że może tutaj powinniśmy wykorzystać tego JSONa dotyczącego logowania, bo on przekazuje informację czy użytkownik jest zalogowany. Czyli w sygnaturze metody oprócz podawania ID usera dodać jeszcze body z tym JSONem login i sprawdzać czy to pole loggedIn ma wartość true

Reply:
Wydaje mi się ze za sekcje logowania i weryfikacje uzytkownika powinna byc dedykowana klasa która bedzie tym zarządzać. Nie wrzucałbym tego do UserControllera.

Reply:
Tak, tak, zgadzam się żeby nie dodawać tego "bezpośrednio" tutaj. Można by to dorzucić do tej klasy UserAuthenticator, którą stworzyłam przy okazji implementacji CartControllera. Mam tylko na myśli to żeby tutaj w parametrze przekazywać tego jsona z logowaniem i przed wygenerowaniem klucza wywoływać metodę, która sprawdzi czy ten user jest np. zalogowany i ma aktywne konto